src/status_im/contexts/wallet/create_account/view.cljs

```diff
@@ -22,14 +22,15 @@
 
 
 (defn- get-keypair-data
-  [primary-name derivation-path account-color {:keys [keypair-name]}]
-  [{:title             (or keypair-name (i18n/label :t/keypair-title {:name primary-name}))
-    :image             (if keypair-name :icon :avatar)
-    :image-props       (if keypair-name
-                         :i/seed
-                         {:full-name           (utils.string/get-initials primary-name 1)
+  [title primary-keypair? derivation-path account-color {:keys [keypair-name]}]
```

Comment:
is there any particular reason we're mixing positional args and a map here? can't it all just be a map, not particularly a fan of api's like this 👍

Reply:
also I wonder would it be easier to read to just make two smaller functions?

```
(defn navigate-to-keypair-page [] (rf/dispatch [:navigate-to :wallet-select-keypair]) )

(defn- get-keypair-data
   [title {:keys [keypair-name]}]
   [{:title             (or keypair-name title)
     :image              :icon
     :image-props        :i/seed
     :action            (when-not keypair-name :button)
     :action-props      {:on-press  navigate-to-keypair-page }}])

  (defn- get-primary-keypair-data
   [title account-color {:keys [keypair-name]}]
   [{:title             (or keypair-name (i18n/label :t/keypair-title {:name title}))
     :image             :avatar
     :image-props       {:full-name           (utils.string/get-initials title 1)
                           :size                :xxs
                           :customization-color account-color}
     :action            (when-not keypair-name :button)
     :action-props      {:on-press   navigate-to-keypair-page }}])

```

Reply:
@J-Son89 I think there will be some duplications. I have simplified the current function to make it more readable.

Reply:
thanks @OmarBasem - duplication is not always a bad thing :) but yeah can be hard to quantify what we value more. My general preference is for readability so whatever makes that easiest, although duplication can add noise to that too 😆

Reply:
It can go both ways, but I think reducing duplication here will reduce the need for duplicated changes in the future also